src/catch2/catch_test_macros.hpp

```diff
@@ -137,7 +137,7 @@
   #define CHECK_THROWS_AS( expr, exceptionType ) INTERNAL_CATCH_THROWS_AS( "CHECK_THROWS_AS", exceptionType, Catch::ResultDisposition::ContinueOnFailure, expr )
   #define CHECK_NOTHROW( ... ) INTERNAL_CATCH_NO_THROW( "CHECK_NOTHROW", Catch::ResultDisposition::ContinueOnFailure, __VA_ARGS__ )
 
-  #define TEST_CASE( ... ) INTERNAL_CATCH_TESTCASE( __VA_ARGS__ )
+  #define TEST_CASE( ... ) INTERNAL_CATCH_TESTCASE( __VA_ARGS__ ) /* NOLINT(cppcoreguidelines-avoid-non-const-global-variables) */
```

Comment:
1) This is not enough, you either need to add this to `CATCH_TEST_CASE` as well, or place it into `INTERNAL_CATCH_TESTCASE`.
2) Why not actually make the variable const? https://github.com/catchorg/Catch2/commit/986c78016e696d29e09abe9b4b7c855bb4f871b5

Reply:
I merged a change that makes all possible registrars `const`. See if this fixes your issue.